WcaOnRails/app/controllers/api/internal/v1/permissions_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Api::Internal::V1::PermissionsController < Api::Internal::V1::ApiController
+  def index
+    user_id = params.require(:user_id)
+    user = User.find(user_id)
+    render json: {
+      can_attend_competitions: {
+        scope: user.cannot_register_for_competition_reasons.empty? ? "*" : [],
+        until: user.banned? ? user.current_team_members.select(:team == Team.banned).first.end_date : nil
```

Comment:
First, you probably want `current_team_members.select(team: Team.banned)`. Also, `select` on Rails ORM collections are just an alias to `where`, which in turn is closer to the actual SQL syntax.

Second, I feel like there _should_ be a shorthand for the "until when is this competitor banned?" information... But of course I don't know all 1000+ lines of the User class by heart so I might be wrong.

Reply:
I took this from the banned_at_date? function in user.rb
```
def banned_at_date?(date)
    if banned?
      ban_end = current_team_members.select(:team == Team.banned).first.end_date
      !ban_end.present? || date < ban_end
    else
      false
    end
  end
```

Reply:
I could make this a new function in the model though if we use it twice now

Reply:
made a ban_end function in the model